trucks.js

```diff
@@ -501,4 +513,176 @@ var foodTrucks = [
+    if ((currentTruck.schedule) && (currentTruck.schedule.length > 0)) {
+      // Use the Array.prototype.find function to find the specified 'day',
+      // ignoring case, in the current truck's schedule.
+      dayMatched = currentTruck.schedule.find(function (currentDay) {
+        return dayRegExpr.test(currentDay);
+      });
+
+      // If there is a match, dayMatched contains the value of the
+      // matched element in the current truck's schedule array. Otherwise,
+      // dayMatched contains 'undefined'.
+      if (dayMatched) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+  });
+
+  return trucks;
+};
+
+/*************************************************************************
+* Description : filterTrucksByFoodType(foodType) - return trucks with
+*               associated 'foodType'
+* Parameters  : foodType - a string
+* Return      : Array of trucks that sell 'foodType'
+*************************************************************************/
+var filterTrucksByFoodType = function (foodType) {
+  var trucks = [];
+  var foodTypeMatched;
+  // Set foodTypeRegExpr to '/value of foodType/i' to search for
+  // 'value of foodType' ignoring case.
+  var foodTypeRegExpr = new RegExp(foodType, 'i');
+
+  trucks = foodTrucks.filter(function (currentTruck) {
+    if ((currentTruck.type) && (currentTruck.type.length > 0)) {
+      foodTypeMatched = currentTruck.type.find(function (currentFoodType) {
+        return foodTypeRegExpr.test(currentFoodType);
+      });
+    }
+    if (foodTypeMatched) {
+      return true;
+    } else {
+      return false;
+    }
+  });
+
+  return trucks;
+};
+
+/*************************************************************************
+* Description : addTruck(objTruck) - This method takes one parameter, a
+*               truck object, and adds it to the foodTrucks array.
+* Parameters  : objTruck - a truck object
+* Return      : the truck object if the operation is successful;
+*               otherwise, null is returned.
+*************************************************************************/
+var addTruck = function (objTruck) {
+  var result = null;
+
+  if ((objTruck !== null) && (typeof objTruck === 'object')) {
+    foodTrucks.push(objTruck);
+    result = objTruck;
+  }
+
+  return result;
+};
+
+/*************************************************************************
+* Description : removeTruck(name) - This method takes one parameter, a
+*               name associated with a food truck. It should remove the
+*               given truck
+* Parameters  : name - a string
+* Return      : 'true' if the operation is successful; 'false' otherwise
+*************************************************************************/
+var removeTruck = function (name) {
+  var result = false;
+  var truckToBeRemoved = getTruck(name);
+
+  if (truckToBeRemoved !== null) {
+    var idx = foodTrucks.indexOf(truckToBeRemoved);
```

Comment:
This won't do what you think it does - check the value you're getting for idx when this function is called. Instead of using indexOf to find objects, you want to find where something specific matches, like the name. See the documentation for the array method `findIndex` for a better way to get the index you're looking for. You'll also find that while it will look like the delete works when you delete a truck, refreshing the page will show that you're always only deleting the last one in the list. This is because your call to indexOf is always returning -1, and supplying a negative value for the index starts it counting from the end of the list.